WordPress/gutenberg
# Extract the pattern overrides toolbar indicator from the `block-editor` package

## What?

Part of https://github.com/WordPress/gutenberg/issues/60619. Address the issue discussed in https://github.com/WordPress/gutenberg/pull/62317#discussion_r1631024218.

Extract out pattern overrides related code from the `block-editor` package to the `patterns` package, specifically the toolbar indicator.

## Why?

Pattern Overrides is a WP-only feature so ideally it doesn't belong in the `block-editor` package.

## How?

Create a `PatternOverridesBlockControls` component and conditionally render it using the pattern-overrides hook.

## Testing Instructions




1. Create a pattern with overrides and insert it into a post.
2. Select an overridable block and expect the toolbar indicator to show (see below image).
3. Select multiple overridable blocks and expect the toolbar indicator to show (see below image).
4. Selecting blocks inside the pattern focus mode should show a different indicator (see below image).

## Screenshots or screencast 

Select a single pattern overrides block:
![image](https://github.com/WordPress/gutenberg/assets/7753001/a5188b6b-c784-47e8-adaf-78a3cd5cb95a)

Select multiple pattern overrides blocks:
![image](https://github.com/WordPress/gutenberg/assets/7753001/e694f815-075e-4f7e-a567-231e0518e184)

Select a block inside the pattern focus mode:
![image](https://github.com/WordPress/gutenberg/assets/7753001/caae75b2-4284-45d2-a988-35a3dcfbbe1e)
